### PR TITLE
Migrate Xcode projects last version checks to Xcode 14.3

### LINKED ISSUE #125817: Newly created iOS project has "Update to recommended settings" warning in Xcode 14.3
Screenshot 2023-05-01 at 11 03 24 AM

The two build settings suggestions should be ignored.  We support iOS 11 and we still need the ignore the quoted header includes https://github.com/flutter/flutter/issues/60025.

The "enable parallelization" is probably good to turn on and test.

```diff
+				BuildIndependentTargetsInParallel = YES;
```
Screenshot 2023-05-01 at 11 13 09 AM

See #90304 for example of this being done for Xcode 13.

### PR DESCRIPTION
**Pull request by jmagman**
1. Add iOS and macOS migration to mark "last upgraded" Xcode version to 14.3 to prevent `Update to recommended settings` warning.
2. Update iOS and macOS templates to same.
3. Update iOS template to set `BuildIndependentTargetsInParallel` to YES as suggested.  I didn't add a migration for this since it seems like a minor optimization and I don't think it's worth a potentially botched/corrupted migration.
4. Run all example/integration test project to see migrator work.
5. Add some missing test projects to the build shard since I noticed they were missing and I had to build those manually outside `SHARD=build_tests`.

Fixes https://github.com/flutter/flutter/issues/125817
See https://github.com/flutter/flutter/pull/90304 for Xcode 13 example.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: dev/bots/test.dart [hunk 1]
```diff
@@ -543,7 +543,10 @@ Future<void> _runBuildTests() async {
     ..add(Directory(path.join(flutterRoot, 'dev', 'integration_tests', 'hybrid_android_views')))
     ..add(Directory(path.join(flutterRoot, 'dev', 'integration_tests', 'flutter_gallery')))
     ..add(Directory(path.join(flutterRoot, 'dev', 'integration_tests', 'ios_platform_view_tests')))
+    ..add(Directory(path.join(flutterRoot, 'dev', 'integration_tests', 'ios_app_with_extensions')))
     ..add(Directory(path.join(flutterRoot, 'dev', 'integration_tests', 'non_nullable')))
+    ..add(Directory(path.join(flutterRoot, 'dev', 'integration_tests', 'platform_interaction')))
+    ..add(Directory(path.join(flutterRoot, 'dev', 'integration_tests', 'spell_check')))
```
Comment (jmagman) [inline dev/bots/test.dart:549]: Add these integration test projects to build in presubmit.

File: packages/flutter_tools/lib/src/migrations/xcode_project_object_version_migration.dart [hunk 2]
```diff
@@ -38,10 +38,10 @@ class XcodeProjectObjectVersionMigration extends ProjectMigrator {
       // objectVersion value has been 46, 50, 51, and 54 in the template.
       RegExp(r'objectVersion = \d+;'): 'objectVersion = 54;',
       // LastUpgradeCheck is in the Xcode project file, not scheme file.
-      // Value has been 0730, 0800, 1020, and 1300 in the template.
-      RegExp(r'LastUpgradeCheck = \d+;'): 'LastUpgradeCheck = 1300;',
+      // Value has been 0730, 0800, 1020, 1300, and 1430 in the template.
+      RegExp(r'LastUpgradeCheck = \d+;'): 'LastUpgradeCheck = 1430;',
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/migrations/xcode_project_object_version_migration.dart:42]: This is the migration change.

File: packages/flutter_tools/templates/app_shared/ios-objc.tmpl/Runner.xcodeproj/project.pbxproj.tmpl [hunk 3]
```diff
@@ -186,7 +186,8 @@
 		97C146E61CF9000F007C117D /* Project object */ = {
 			isa = PBXProject;
 			attributes = {
-				LastUpgradeCheck = 1300;
+				BuildIndependentTargetsInParallel = YES;
```
Comment (jmagman) [inline packages/flutter_tools/templates/app_shared/ios-objc.tmpl/Runner.xcodeproj/project.pbxproj.tmpl:189]: Added to iOS template but not migration.

File: packages/flutter_tools/templates/app_shared/macos.tmpl/Runner.xcodeproj/project.pbxproj.tmpl [hunk 4]
```diff
@@ -227,7 +227,7 @@
 			isa = PBXProject;
 			attributes = {
 				LastSwiftUpdateCheck = 0920;
-				LastUpgradeCheck = 1300;
+				LastUpgradeCheck = 1430;
```
Comment (jmagman) [inline packages/flutter_tools/templates/app_shared/macos.tmpl/Runner.xcodeproj/project.pbxproj.tmpl:230]: Not sure if `BuildIndependentTargetsInParallel` was needed for macOS template, there was no prompt in Xcode to turn it on.  Skip it.

**Review by vashworth — APPROVED** [review]
LGTM

**Review by christopherfujino — APPROVED** [review]
LGTM

File: dev/integration_tests/spell_check/ios/Runner.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -7,6 +7,7 @@
 	objects = {
 
 /* Begin PBXBuildFile section */
+		08DF46C1ECE03723F402A526 /* libPods-Runner.a in Frameworks */ = {isa = PBXBuildFile; fileRef = 49B50C6BD3C1A11B60BCFF13 /* libPods-Runner.a */; };
```
[the remark this answers is not in the document]
Comment (jmagman) [reply dev/integration_tests/spell_check/ios/Runner.xcodeproj/project.pbxproj:10]: It looks like `spell_check` wasn't checked in after being built?  Anyway I built it and CocoaPods and Xcode wanted this.

**Review by cyanglaz — APPROVED** [review]
LGTM

File: dev/integration_tests/spell_check/ios/Runner.xcodeproj/project.pbxproj (hunk 5 above)
[the remark this answers is not in the document]
Comment (christopherfujino) [reply dev/integration_tests/spell_check/ios/Runner.xcodeproj/project.pbxproj:10]: word

**Pull request merged**